File: App/public_html/app/services/EmployeeService.js
/**
 * Created by Negin Nafissi on 13-12-2016.
 */

angular.module('IN2').service('employeeService', function ($http, alertify) {

    var self = this;

    self.create = function (firstName, lastName, study, birthDate, email, phoneNumber, tag, address, cityName, zipCode, noteText, onCreated) {

        var uri = '/api/employees';
        var data =
            {
                firstName: firstName,
                lastName: lastName,
                study: study,
                birthDate: birthDate,
                email: email,
                phoneNumber: phoneNumber,
                tag: tag,
                address: address,
                city: cityName,
                postcode: zipCode,
                noteText: noteText
            };
        $http.post(uri, data).then(function (response) {
                onCreated(response.data);
            },
            function () {
                alertify.okBtn("OK").confirm("Aanmaken mislukt, neem contact op met uw beheerder!");
            });
    };

    self.getAll = function (onReceived) {

        var uri = 'api/employees';

        $http.get(uri).then(function (response) {
                onReceived(response.data);
            },
            function () {
                alertify.okBtn("OK").confirm("Ophalen mislukt, neem contact op met uw beheerder!");
            });

    };

    self.selectedEmployee = 0;

    self.setSelected = function (id) {
        self.selectedEmployee = id;
    };

    self.getEmployee = function (onReceived) {
        var uri = 'api/employees/' + self.selectedEmployee + '';

        $http.get(uri).then(function (response) {
                onReceived(response.data);
            },
            function () {
                alertify.okBtn("OK").confirm("Ophalen mislukt, neem contact op met uw beheerder!");
            });
    };

    self.update = function (employee, onReceived) {
        var uri = 'api/employees/' + employee.id + '';

        $http.put(uri, employee).then(function (response) {
                onReceived(response.data);
            },
            function () {
                alertify.okBtn("OK").confirm("Aanpassen mislukt, neem contact op met uw beheerder!");
            });
    };

    self.delete = function (id, onReceived) {
        var uri = 'api/employees/' + id + '';

        $http.delete(uri, id).then(function (response) {
                onReceived(response.data);
                self.getAll(onReceived);
            },
            function () {
                alertify.okBtn("OK").confirm("Verwijderen mislukt, neem contact op met uw beheerder!");
            });
    };

});

File: App/Server/Service/EmployeeService.java
package Server.Service;

import Server.Model.Employee;
import Server.Persistence.EmployeeDAO;

import javax.inject.Inject;
import java.util.Collection;

/**
 * Created by Negin Nafissi on 13-12-2016.
 */
public class EmployeeService extends BaseService<Employee> {

    private final EmployeeDAO dao;

    /**
     * Instantiates a new Employee service.
     *
     * @param dao the dao
     */
    @Inject
    public EmployeeService(EmployeeDAO dao){
        this.dao = dao;
    }

    /**
     * Get all collection.
     *
     * @return the collection
     */
    public Collection<Employee> getAll(){
        return dao.getAll();
    }

    /**
     * Get employee.
     *
     * @param id the id
     * @return the employee
     */
    public Employee get(int id){
        return dao.getEmployee(id);
    }

    /**
     * Add.
     *
     * @param employee the employee
     */
    public void add(Employee employee){
        dao.addEmployee(employee);
    }

    /**
     * Update.
     *
     * @param employee the employee
     */
    public void update(Employee employee) {
        dao.update(employee);
    }

    /**
     * Delete.
     *
     * @param id the id
     */
    public void delete(int id) {
        dao.delete(id);
    }


}
